gobblin-config-management/src/main/java/gobblin/config/utils/URIComparator.java

```diff
@@ -0,0 +1,56 @@
+package gobblin.config.utils;
+
+import java.net.URI;
+import java.util.Comparator;
+
+
+public class URIComparator implements Comparator<URI> {
```

Comment:
This comparator is not consistent with equals and it is also not anti-symmetric (e.g. a > b does not necessarily imply b < a). This can lead to strange bugs. I suggest using the natural ordering of URIs, and instead using the method TreeMap.floorKey to determine ascendance.